src/reader.jl

```diff
@@ -402,6 +402,10 @@ function read_field_(reader::AbstractBSONReader, ::Type{Vector{T}}) where T
     copy!(dst, reader)
 end
 
+function read_field_(reader::T, ::Type{<:Union{T,AbstractBSONReader}}) where {T<:AbstractBSONReader}
```

Comment:
Just some spacing for uniformity with the rest
```suggestion
function read_field_(reader::T, ::Type{<:Union{T, AbstractBSONReader}}) where {T <: AbstractBSONReader}
```